### PR TITLE
[Network] Add support for --dns-servers to NIC create/update and VNET update.

### LINKED ISSUE #3558: [Network] NIC create/update should support custom DNS servers
NIC creation currently supports only `--internal-dns-name`. However, there are other DNS properties that can be set via the portal. The CLI should support these scenarios as well.

### LINKED ISSUE #3615: [Network] Expose DNS Servers convenience argument for VNET create
Xplat has this argument exposed, and we expose it on create.

### PR DESCRIPTION
**Pull request by tjprescott**
Closes #3558. Closes #3615. Partial fix for #3386.

@troydai I tried converting this to a scenario test, but ran into bizarre behavior when I tried to record the test. When I record the test with the old test base, it works fine, so I kept it as is.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [X] Each command and parameter has a meaningful description.
- [X] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by tjprescott** [comment]
DNM temporarily while I add the commits to fix #3615.

**Comment by tjprescott** [comment]
Added support to VNET create and enhanced the test to cover the parameter behavior.

**Comment by troydai** [comment]
@tjprescott please show me what's the issue with `ScenarioTest` tomorrow.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -1052,6 +1054,10 @@ def update_nic(instance, network_security_group=None, enable_ip_forwarding=None,
         instance.dns_settings.internal_dns_name_label = None
     elif internal_dns_name_label is not None:
         instance.dns_settings.internal_dns_name_label = internal_dns_name_label
+    if dns_servers == ['']:
```
Comment (troydai) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py]: This is actually what network SDK will do? What a bizarre behavior.

**Comment by tjprescott** [comment]
@troydai the issue recurs even when I use the LiveTest base. I would like to go ahead and merge this PR and track this as a separate issue with the testing infrastructure.

**Pull request merged**